package.json

```diff
@@ -24,22 +24,24 @@
         "cueball": "2.2.9",
         "dashdash": "1.3.2",
         "deep-equal": "0.0.0",
-        "dtrace-provider": "0.2.8",
+        "dtrace-provider": "0.8.8",
         "fast": "2.8.1",
         "http-signature": "1.1.0",
         "jsprim": "^1.3.1",
         "kang": "1.1.0",
         "keep-alive-agent": "0.0.1",
         "keyapi": "git+https://github.com/joyent/keyapi.git#e14b3d58",
         "libmanta": "git+https://github.com/joyent/node-libmanta.git#v1.3.0",
```

Comment:
There are a number of `gyp ERR! build error` for, I'm guessing,
optionalDependencies in the build lot. E.g. for old bunyan:


```
...
   ^
../dtrace_provider.cc: In static member function 'static v8::Handle<v8::Value> node::DTraceProvider::Fire(const int&)':
../dtrace_provider.cc:186:3: warning: control reaches end of non-void function [-Wreturn-type]
   }
   ^
DTraceProviderBindings.target.mk:99: recipe for target 'Release/obj.target/DTraceProviderBindings/dtrace_provider.o' failed
make[1]: *** [Release/obj.target/DTraceProviderBindings/dtrace_provider.o] Error 1
make[1]: Leaving directory '/root/joy/manta-buckets-api/node_modules/bunyan/node_modules/dtrace-provider/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/root/joy/manta-buckets-api/build/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:219:12)
gyp ERR! System SunOS 5.11
gyp ERR! command "/root/joy/manta-buckets-api/build/node/bin/node" "/root/joy/manta-buckets-api/build/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /root/joy/manta-buckets-api/node_modules/bunyan/node_modules/dtrace-provider
gyp ERR! node -v v6.17.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
```

and for libmanta issues with node v6:

```
DTraceProviderBindings.target.mk:99: recipe for target 'Release/obj.target/DTraceProviderBindings/dtrace_provider.o' failed
make[1]: *** [Release/obj.target/DTraceProviderBindings/dtrace_provider.o] Error 1
make[1]: Leaving directory '/root/joy/manta-buckets-api/node_modules/libmanta/node_modules/restify/node_modules/dtrace-provider/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/root/joy/manta-buckets-api/build/node/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:219:12)
gyp ERR! System SunOS 5.11
gyp ERR! command "/root/joy/manta-buckets-api/build/node/bin/node" "/root/joy/manta-buckets-api/build/node/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /root/joy/manta-buckets-api/node_modules/libmanta/node_modules/restify/node_modules/dtrace-provider
gyp ERR! node -v v6.17.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
```

I think these should probably be sorted first. The libmanta one might be a real
pain. I don't have much experience, but libmanta is a grabbag of tons of shared
crap. That it has its own *restify* is curious. Finding a shared restify version
that works for all users of libmanta might be "fun".

It is possible that cleaning libmanta up might be a part of the work here.
It is also possible that cleaning that up might be a lot easier once we have
the old mantav1 stuff branched off.

Reply:
Our usage of `libmanta` in `buckets-api` is pretty minimal and in fact I'd like to remove our usage of it altogether. There's just three functions we use and those could just be imported to the `buckets-api` `common` module.

Reply:
+1 to dropping that dep then. It could very well be a node v6 mine field.  It really mostly feels (from my limited exposure) more like a "libmantadirstyle", if you know what I mean.

Reply:
I'm going to look into this today

Reply:
created MANTA-4764... if you both are cool with it, i can tackle this outside of the upgrade to node v6 since it's not currently blocking

Reply:
There are still "gyp: ERR!" build errors for:

- libmanta: I'm fine with defering that to MANTA-4764 as long as that doesn't linger too long
- watershed: Is there an updated watershed version with more modern dtrace-provider?
- smartdc: A devDependency. Does updating to smartdc@9.0.0 fix that? It looks like it might from this change:

```
commit db6910d25216e86658f82a80cfa1878664bf2721
Author: Brian Bennett
Date:   2015-10-22T21:23:25-07:00 (4 years, 1 month ago)

    Update dep to get newer dtrace-provider to fix build against node 4.x

diff --git a/package.json b/package.json
index 805ed18..05501c5 100644
--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "author": "Joyent, Inc. http://www.joyent.com",
   "name": "smartdc",
   "description": "Client SDK and CLI for the Joyent SmartDataCenter API",
-  "version": "7.5.3",
+  "version": "7.5.4",
```

Reply:
1. I'll take an MANTA-4764 and will get it done soon (before december)
2. good news! `watershed` isn't actually used... so let's remove it
3. i'll update to 9.0.0 and test it

```
$ rg watershed
package.json
48:        "watershed": "0.3.0",
```

^^ that's all that showed up

Reply:
Great. 

FWIW, looks like the same for `strsplit` and `deep-equal` as well.  Checked via:

```
json -f package.json dependencies | json -Ma key \
    | while read dep; do echo ""; echo "# $dep"; rg $RG_OPTS --no-heading "require\('$dep['/]" </dev/null; done
json -f package.json devDependencies | json -Ma key \
    | while read dep; do echo ""; echo "# $dep"; rg $RG_OPTS --no-heading "require\('$dep['/]" </dev/null; done
```

Reply:
@trentm good catch! i'm removing them now.  I'd love to see some form of that check in `make check` or something like that.  like `dependency "XXX" is unused`

Reply:
>  I'd love to see some form of that check in make check or something like that. like dependency "XXX" is unused

Might be hard to make that reliable. There are some deps that are valid for using their binaries (./node_modules/.bin/foo), but not `require`d.